### DISCUSSION TITLE
Save Data

### DISCUSSION THREAD

**Discussion by FireVirusMC** [EmulatorJS/EmulatorJS#400] [Q&A]
Is it possible to grab save data from a game you are playing to edit or transfer onto a game catridge?

**Comment by ethanaobrien** [comment]
No, this is not and will not be possible. This would also be the kind of issue you open in the retroarch repository itself. I'm not that advanced (and don't have the tools to do that)

**Reply by FireVirusMC** [reply]
heh, all good I'll just replay the game on my ds. Don't know why I didn't do that. Thanks for your hard work bro.

**Comment by allancoding** [comment]
Look at this video: https://www.youtube.com/watch?v=YGmo3yiRYeQ

**Reply by FireVirusMC** [reply]
Hey Allan, thanks for the advice, but I was looking to extract the save data from emulatorJS and then import it into my game card.

**Reply by allancoding** [reply]
What system are you trying to do it with?

**Reply by FireVirusMC** [reply]
the gameboy advance sp

**Reply by allancoding** [reply]
Use: https://savefileconverter.com/#/gba/gameshark-sp To convert the file

Then look here:
how can I [copy save files to and from a GBA cartridge](https://savefileconverter.com/#/original-hardware?sort=gba)?
how do I [copy save files to and from my GameShark SP](https://www.manualslib.com/manual/515605/Mad-Catz-Gameshark-Sp.html?page=17#manual)?

**Reply by FireVirusMC** [reply]
this is amazing, but the pre requisite for this is to have the save rom for the emulator and then convert it into a save file that can be put on the gba. I found an alternate method to make this process a little easier to do it on my 3ds first and export the gba file to put on my gba. Still as ethan said it is probably impossible to export the save file from emulatorJS itself without using retroarch.

**Reply by allancoding** [reply]
EmulatorJS uses retroarch cores. I have tested this I saved a game on retroarch and loaded it to EmulatorJS and I continued in the game where I left of in retroarch, I then did somethings in the game saved in EmulatorJS and put in in my retroarch's save dir and continued playing the game.

**Reply by FireVirusMC** [reply]
this is amazing, thank you.

**Reply by ethanaobrien** [reply]
#377 is probably what you're asking for, am I correct

**Reply by FireVirusMC** [reply]
yes, exactly I've converted the .srm file I obtained from putting my save state into retroarch to a .sav and since I'm not as advanced enough to inject the .sav into my GBA cartridge ( and I dont have the materials ) I decided to inject the save file into my GBA virtual console injection on my 3ds. It works properly. Like in #377 I highly recommend adding a feature to dump the save file. (Unless it is already a feature)
